server/neptune/gateway/event/pull_request_review_handler.go

```diff
@@ -0,0 +1,87 @@
+package event
+
+import (
+	"bytes"
+	"context"
+	"github.com/runatlantis/atlantis/server/lyft/feature"
+
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/http"
+	"github.com/runatlantis/atlantis/server/logging"
+)
+
+const (
+	Approved = "APPROVED"
+)
+
+type PullRequestReview struct {
+	InstallationToken int64
+	Repo              models.Repo
+	PRNum             int
+	User              models.User
+	Action            string
+	Ref               string
+}
+
+type fetcher interface {
+	ListFailedPolicyCheckRuns(ctx context.Context, installationToken int64, repo models.Repo, ref string) ([]string, error)
+}
+
+type PullRequestReviewWorkerProxy struct {
+	Scheduler       scheduler
+	SnsWriter       Writer
+	Logger          logging.Logger
+	Allocator       feature.Allocator
+	CheckRunFetcher fetcher
+}
+
+func (p *PullRequestReviewWorkerProxy) Handle(ctx context.Context, event PullRequestReview, request *http.BufferedRequest) error {
+	return p.Scheduler.Schedule(ctx, func(ctx context.Context) error {
+		return p.handle(ctx, event, request)
+	})
+}
+
+func (p *PullRequestReviewWorkerProxy) handle(ctx context.Context, event PullRequestReview, request *http.BufferedRequest) error {
+	shouldAllocate, err := p.Allocator.ShouldAllocate(feature.PolicyV2, feature.FeatureContext{
+		RepoName: event.Repo.FullName,
+	})
+	if err != nil {
+		p.Logger.ErrorContext(ctx, "unable to allocate policy v2")
+		return err
+	}
+	if !shouldAllocate {
+		return nil
+	}
+
+	// Ignore non-approval events
+	if event.Action != Approved {
+		return nil
+	}
+
+	// Ignore PRs without failing policy checks
+	failedPolicyCheckRuns, err := p.CheckRunFetcher.ListFailedPolicyCheckRuns(ctx, event.InstallationToken, event.Repo, event.Ref)
+	if err != nil {
+		p.Logger.ErrorContext(ctx, "unable to allocate policy v2")
+		return err
+	}
+	if len(failedPolicyCheckRuns) == 0 {
+		p.Logger.ErrorContext(ctx, "no failed policies, skipping event")
+		return nil
+	}
+	// Forward to SNS to further process in the worker
+	return p.forwardToSns(ctx, request)
```

Comment:
I'm assuming the reason we're doing this is because we need access to the db on the worker right?

Reply:
yes, since PR mode still uses the legacy structure, the db needs to store when the state of the policy check